From 0c59837d920b4d6cb56577dbc3b565508ee5361d Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Tue, 26 May 2020 16:40:41 +0200
Subject: [PATCH] devicetree, miamimp: control VBUS power

Enable VBUS power only when USB is in host mode
---
 .../arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
index 35f0e0f6e531..2b3ed5644ec5 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
@@ -22,7 +22,6 @@
 		regulator-max-microvolt = <5000000>;
 		gpio = <&gpioexf 1 0>;
 		enable-active-high;
-		regulator-always-on; /* for USB host mode, "vbus" no longer works? */
 	};
 
 	xtal_27Mhz: clock_xtal_27Mhz {
@@ -212,6 +211,7 @@
 &dwc3_0 {
 	status = "okay";
 	dr_mode = "otg"; /* "host", "peripheral" or "otg" */
+	vbus-supply = <&reg_usb0_vbus>;
 	maximum-speed = "high-speed"; /* super-speed not available on carrier */
 	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
 	snps,dis_u2_susphy_quirk;
@@ -219,7 +219,6 @@
 	/delete-property/ phy-names;
 	/delete-property/ phys;
 	usb-phy = <&usb_phy0>;
-	vbus-supply = <&reg_usb0_vbus>; /* This apparently no longer works */
 	ulpi {
 		phy {
 			compatible = "smsc,usb3340";
-- 
2.17.1

